# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/util-linux/hotfixes.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -dur util-linux-2.12h/lib/nls.h util-linux-2.12h-patch/lib/nls.h
--- util-linux-2.12h/lib/nls.h	2002-07-04 21:03:26.000000000 +0000
+++ util-linux-2.12h-patch/lib/nls.h	2004-10-23 03:17:42.544527264 +0000
@@ -1,6 +1,15 @@
 int main(int argc, char *argv[]);
 
 #include "../defines.h"		/* for HAVE_locale_h */
+#ifndef HAVE_locale_h
+#  define HAVE_locale_h 1
+#endif
+
+#ifndef HAVE_langinfo_h
+#  define HAVE_langinfo_h 1
+#endif
+
+
 
 #ifndef PACKAGE
 #define PACKAGE	"util-linux"
diff -dur util-linux-2.12h/MCONFIG util-linux-2.12h-patch/MCONFIG
--- util-linux-2.12h/MCONFIG	2004-09-16 16:30:47.000000000 +0000
+++ util-linux-2.12h-patch/MCONFIG	2004-10-23 03:18:22.577441336 +0000
@@ -6,6 +6,8 @@
 #  - remove the `-o root' part in INSTALLSUID
 #  - set USE_TTY_GROUP=no
 #  - define DESTDIR
+# compile the raw utility
+ADD_RAW=yes
 
 ## Configuration outside of this file you might want to do for mount:
 ## If make_include has HAVE_BLKID=yes, turn that into HAVE_BLKID=no
@@ -16,8 +18,8 @@
 # Select for ARCH one of intel, alpha, sparc, arm, m68k, mips
 # Select for CPU i386 if the binaries must be able to run on an intel 386
 # (by default i486 code is generated, see below)
-CPU=$(shell uname -m)
-ARCH=$(shell echo $(CPU) | sed 's/i.86/intel/;s/arm.*/arm/')
+# CPU=$(shell uname -m)
+# ARCH=$(shell echo $(CPU) | sed 's/i.86/intel/;s/arm.*/arm/')
 
 # If HAVE_PAM is set to "yes", then login, chfn, chsh, and newgrp
 # will use PAM for authentication. Additionally, passwd will not be
@@ -125,8 +127,7 @@
   else
     CPUTAIL=486
   endif
-  CPUOPT= $(CPUHEAD)$(CPUTAIL)
-  OPT=          -pipe -O2 $(CPUOPT) -fomit-frame-pointer
+  OPT=          -pipe -O2 -fomit-frame-pointer
 else 
   ifeq "$(ARCH)" "arm"
     OPT=	-pipe -O2 -fsigned-char -fomit-frame-pointer
Nur in util-linux-2.12h-patch/: MCONFIG~.
Nur in util-linux-2.12h-patch/: MCONFIG.rej.
diff -dur util-linux-2.12h/mount/mount.c util-linux-2.12h-patch/mount/mount.c
--- util-linux-2.12h/mount/mount.c	2004-10-13 20:54:39.000000000 +0000
+++ util-linux-2.12h-patch/mount/mount.c	2004-10-23 03:17:42.547526808 +0000
@@ -37,7 +37,8 @@
 #include "env.h"
 #include "nls.h"
 
-#define DO_PS_FIDDLING
+// #define DO_PS_FIDDLING  /* this could be a Problem */
+
 
 #ifdef DO_PS_FIDDLING
 #include "setproctitle.h"
Nur in util-linux-2.12h-patch/mount: mount.c~.
diff -dur util-linux-2.12h/text-utils/Makefile util-linux-2.12h-patch/text-utils/Makefile
--- util-linux-2.12h/text-utils/Makefile	2003-03-02 23:32:04.000000000 +0000
+++ util-linux-2.12h-patch/text-utils/Makefile	2004-10-23 03:17:42.547526808 +0000
@@ -44,6 +44,7 @@
 # Do not have ncurses - give up on pg and ul
 pg ul:
 	@echo $@ not made since it requires ncurses
+	@touch $@
 # For more we can also try termcap
 ifeq "$(HAVE_TERMCAP)" "yes"
 more: more.o $(LIB)/xstrncpy.o
@@ -51,6 +52,7 @@
 else
 more:
 	@echo $@ not made since it requires ncurses or termcap
+	@touch $@
 endif
 endif
 
Nur in util-linux-2.12h-patch/text-utils: Makefile~.
